janus_core/calculations/geom_opt.py

```diff
@@ -233,6 +242,15 @@ def __init__(
         if not self.struct.calc:
             raise ValueError("Please attach a calculator to `struct`.")
 
+        # Warn for deprecated filter_func now that logging is set up
+        if filter_func:
+            warnings.warn(
+                "`filter_func` has been deprecated, but currently overrides "
+                "`filter_class`. Please only set `filter_class`.",
+                FutureWarning,
```

Comment:
https://docs.python.org/3.11/library/exceptions.html#DeprecationWarning

Reply:
Surely this is intended to be seen by end users?

> exception FutureWarning
> Base class for warnings about deprecated features when those warnings are intended for end users of applications that are written in Python.

If I remember correctly, `DeprecationWarning` tended to be hidden by default (unless called via `__main__`), so wasn't really visible.

Reply:
Yep, looks like you're right.